managed/CounterStrikeSharp.API/Modules/Commands/Targeting/Target.cs

```diff
@@ -124,4 +111,24 @@ public TargetResult GetTarget(CCSPlayerController? caller)
         
         return new TargetResult() { Players = players };
     }
+
+    public CCSPlayerController? GetClientAimTarget(CCSPlayerController player)
+    {
+        CCSGameRules? GameRules = Utilities.FindAllEntitiesByDesignerName<CCSGameRulesProxy>("cs_gamerules").FirstOrDefault()?.GameRules;
```

Comment:
this seems slow to me, might be good to find a way to cache gamerules to avoid calling it multiple times for no reason

Reply:
> this seems slow to me, might be a good idea to find a way to cache gamerules to avoid calling it multiple times for no reason

Maybe, something like that?

```csharp
CCSGameRules GameRules { get; set; } = Utilities.FindAllEntitiesByDesignerName<CCSGameRulesProxy>("cs_gamerules").FirstOrDefault()!.GameRules!;

public CCSPlayerController? GetClientAimTarget(CCSPlayerController? player)
{
    if (player == null || player.IsValid)
    {
        return null;
    }

    VirtualFunctionWithReturn<IntPtr, IntPtr, IntPtr> findPickerEntity = new(GameRules.Handle, 28);
    CBaseEntity target = new(findPickerEntity.Invoke(GameRules.Handle, player.Handle));

    if (target.DesignerName is "player")
    {
        return target.As<CCSPlayerPawn>().OriginalController.Value;
    }

    return null;
}
```

Reply:
this should and will fail when a new gamerules entity is created